src/zenml/zen_stores/secrets_stores/aws_secrets_store.py

```diff
@@ -144,46 +193,15 @@ class AWSSecretsStore(BaseSecretsStore):
     workspace) does not update the secret's `updated` timestamp. This is a
     limitation of the AWS Secrets Manager API (updating AWS tags does not update
     the secret's `updated` timestamp).
-
-
-    Attributes:
-        config: The configuration of the AWS secrets store.
-        TYPE: The type of the store.
-        CONFIG_TYPE: The type of the store configuration.
     """
 
     config: AWSSecretsStoreConfiguration
     TYPE: ClassVar[SecretsStoreType] = SecretsStoreType.AWS
     CONFIG_TYPE: ClassVar[
-        Type[SecretsStoreConfiguration]
+        Type[ServiceConnectorSecretsStoreConfiguration]
     ] = AWSSecretsStoreConfiguration
-
-    _client: Optional[Any] = None
-
-    @property
-    def client(self) -> Any:
-        """Initialize and return the AWS Secrets Manager client.
-
-        Returns:
-            The AWS Secrets Manager client.
-        """
-        if self._client is None:
-            # Initialize the AWS Secrets Manager client with the
-            # credentials from the configuration, if provided.
-            self._client = boto3.client(
-                "secretsmanager",
-                region_name=self.config.region_name,
-                aws_access_key_id=self.config.aws_access_key_id.get_secret_value()
-                if self.config.aws_access_key_id
-                else None,
-                aws_secret_access_key=self.config.aws_secret_access_key.get_secret_value()
-                if self.config.aws_secret_access_key
-                else None,
-                aws_session_token=self.config.aws_session_token.get_secret_value()
-                if self.config.aws_session_token
-                else None,
-            )
-        return self._client
+    SERVICE_CONNECTOR_TYPE: ClassVar[str] = AWS_CONNECTOR_TYPE
+    SERVICE_CONNECTOR_RESOURCE_TYPE: ClassVar[str] = AWS_RESOURCE_TYPE
```

Comment:
Does `aws-generic` include checks for accessing the secrets store API?

Reply:
Same question for GCP and Azure.

Reply:
No, `aws-generic` doesn't care about the particular cloud service that the credentials are used to access. We could add another resource type and call it `secrets-manager` for example, and include those checks, but that would take more effort and expose "secrets manager" as a public resource type that the service connectors are capable of managing, like `docker-registry` and `kubernetes-cluster`. However, there's no clear advantage for making this an established service connector feature, because there are no stack components that can be "linked" to a secrets manager (we had the Secrets Manager stack component, but we decided to remove it).

Reply:
Makes sense to not "expose" the secrets manager as a resource, I agree. However, does this mean that we can't verify if the credentials that are provided can access the secrets API? Shouldn't be a big deal in any case, we would get the error from the API anyways.

Reply:
There is some basic verification happening still, but it's mainly to verify that the credentials are valid and can be used to authenticate to the target AWS/GCP/Azure cloud:

* for AWS, we create a boto3 session and call `get_caller_identity`:
```        sts_client = session.client("sts")
        try:
            sts_client.get_caller_identity()
        except (ClientError, BotoCoreError) as err:
            msg = f"failed to verify AWS account access: {err}"
            logger.debug(msg)
            raise AuthorizationException(msg) from err
```

* for GCP, we use the GCP auth library to exchange the credentials for an OAuth2 token:
```
            try:
                with requests.Session() as session:
                    req = Request(session)
                    credentials.refresh(req)
            except google.auth.exceptions.GoogleAuthError as e:
                raise AuthorizationException(
                    f"Could not fetch GCP OAuth2 token: {e}"
                )
```

* for Azure, we don't really verify anything, because the Azure authentication API is the least advanced of the three 

So generally speaking, we do check that the credentials are valid (well formed, can be used to authenticate to the target cloud). However, there is no verification done concerning permissions to access the Secrets Manager API though, not until the user actually tries to create/get/list/delete secrets.

Reply:
Thanks for the detailed response!